hoot-core/src/main/cpp/hoot/core/ops/AddFeatureDimensionsOp.cpp

```diff
@@ -0,0 +1,141 @@
+ *
+ * @copyright Copyright (C) 2015, 2016, 2017, 2018 DigitalGlobe (http://www.digitalglobe.com/)
+ */
+
+#include "AddFeatureDimensionsOp.h"
+
+// geos
+#include <geos/geom/LineString.h>
+#include <geos/geom/MultiLineString.h>
+#include <geos/geom/Polygon.h>
+
+// Hoot
+#include <hoot/core/OsmMap.h>
+#include <hoot/core/schema/OsmSchema.h>
+#include <hoot/core/util/ElementConverter.h>
+#include <hoot/core/util/Factory.h>
+#include <hoot/core/util/Log.h>
+#include <hoot/core/util/MapProjector.h>
+
+using namespace geos::geom;
+
+namespace hoot
+{
+
+HOOT_FACTORY_REGISTER(OsmMapOperation, AddFeatureDimensionsOp)
+
+AddFeatureDimensionsOp::AddFeatureDimensionsOp()
+{
+}
+
+void AddFeatureDimensionsOp::apply(boost::shared_ptr<OsmMap> &map)
+{
+  boost::shared_ptr<OsmMap> planar;
+
+    // If the map is geographic, we need to project it so we can get measurements in metres.
+  if (MapProjector::isGeographic(map))
+  {
+    planar.reset(new OsmMap(map));
+    MapProjector::projectToPlanar(planar);
+  }
+  else
+  {
+    planar = map;
+  }
+
+  // find the area(s) as ways
+  const WayMap& ways = planar->getWays();
+  for (WayMap::const_iterator wayItr = ways.begin(); wayItr != ways.end(); ++wayItr)
+  {
+    const int wayId = wayItr->first;
+
+    ElementPtr pElement = planar->getElement(ElementType::Way, wayId);
+
+    // NOTE: pElement and tElement may be identical
+    ElementPtr tElement = map->getElement(pElement->getElementId());
+
+    boost::shared_ptr<Geometry> g = ElementConverter(planar).convertToGeometry(pElement, true, true);
+
+    if (OsmSchema::getInstance().isArea(pElement))
+    {
+      if (pElement->getTags().get("feature_area").isEmpty())
+      {
+        tElement->getTags()["feature_area"] = QString::number(g->getArea(),'f',2);
+      }
+      continue;
+    }
+
+    // Not an Area, must be Linear
+    if (pElement->getTags().get("length").isEmpty())
+    {
+      tElement->getTags()["length"] = QString::number(g->getLength(),'f',2);
+    }
+  }
+
+  // find the area(s) as Relations
+  const RelationMap& rels = planar->getRelations();
+  for (RelationMap::const_iterator relItr = rels.begin(); relItr != rels.end(); ++relItr)
+  {
+    const int relId = relItr->first;
+
+    ElementPtr pElement = planar->getElement(ElementType::Relation, relId);
+    ElementPtr tElement = map->getElement(pElement->getElementId());
+
+    boost::shared_ptr<Geometry> g = ElementConverter(planar).convertToGeometry(pElement, true, true);
+
+    if (OsmSchema::getInstance().isArea(pElement))
+    {
+      if (pElement->getTags().get("feature_area").isEmpty())
+      {
+        tElement->getTags()["feature_area"] = QString::number(g->getArea(),'f',2);
+      }
+      continue;
+    }
+
+    // Just making sure....
+    if (OsmSchema::getInstance().isLinear(*pElement))
+    {
+      if (pElement->getTags().get("length").isEmpty())
+      {
+        tElement->getTags()["length"] = QString::number(g->getLength(),'f',2);
+      }
+      continue;
```

Comment:
This `continue` doesn't do much, and it can go away. For readability, consider ditching the `continue`s in this file, and exchanging them for a more traditional if-else structure.